### PR TITLE
added bestseller twig function

### PR DESCRIPTION
**Pull request by muammertop**

### CODE REVIEW

File: addons/default/visiosoft/advs-module/src/Adv/AdvModel.php [hunk 1]
```diff
@@ -17,6 +18,51 @@
...
+
+    public function getCategory1Attribute()
+    {
+        return $this->hasMany('Visiosoft\CatsModule\Category\CategoryModel', 'id', 'cat1')->first();
+
+    }
+
+    public function getCategory2Attribute()
+    {
+        return $this->hasMany('Visiosoft\CatsModule\Category\CategoryModel', 'id', 'cat1')->first();
```
Comment (fatihalp) [inline addons/default/visiosoft/advs-module/src/Adv/AdvModel.php:53]: maybe cat2? @muammertop

**Review by fatihalp — CHANGES_REQUESTED** [review]
maybe cat2? @muammertop

**Review by spektra2147 — APPROVED** [review]

File: addons/default/visiosoft/advs-module/src/AdvsModulePlugin.php [hunk 2]
```diff
@@ -49,6 +49,19 @@ function () {
...
+	            'bestsellerAds',
+		        function ($catId = null) {
+	            	return AdvModel::query()
+			            ->orderBy('total_sales', 'desc')
+			            ->where(function ($query) use ($catId) {
+			            	if ($catId) {
+			            		$query->where('cat1', $catId);
+				            }
+			            })
+			            ->limit(10)->get();
```
Comment (diashalabi) [inline addons/default/visiosoft/advs-module/src/AdvsModulePlugin.php:62]: can you move to repo

**Review by diashalabi — CHANGES_REQUESTED** [review]

**Review by spektra2147 — APPROVED** [review]

File: addons/default/visiosoft/advs-module/src/Adv/AdvRepository.php [hunk 3]
```diff
@@ -391,6 +391,17 @@ public function latestAds()
...
 
+	public function bestsellerAds($catId = null, $limit = 10)
+	{
+		return $this->model->orderBy('total_sales', 'desc')
+			->where(function ($query) use ($catId) {
+				if ($catId) {
+					$query->where('cat1', $catId);
+				}
+			})
+			->limit($limit)->get();
```
Comment (diashalabi) [inline addons/default/visiosoft/advs-module/src/Adv/AdvRepository.php:402]: maybe you need to check the status? and if the ad is expired?

File: addons/default/visiosoft/advs-module/src/OptionHandler/AdvsOptions.php [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+	public function __construct(AdvRepositoryInterface $advRepository)
+	{
+		$this->advRepository = $advRepository;
+	}
+
+	public function handle(CheckboxesFieldType $fieldType)
+	{
+		$categories = $this->advRepository->all();
+		$options = $categories->pluck('name', 'id')->all();
+		$fieldType->setOptions($options);
```
Comment (diashalabi) [inline addons/default/visiosoft/advs-module/src/OptionHandler/AdvsOptions.php:19]: what if we had 1,000,000 ads?

**Review by diashalabi — CHANGES_REQUESTED** [review]

File: addons/default/visiosoft/advs-module/src/Adv/AdvRepository.php (hunk 3 above)
Comment (fatihalp) [reply addons/default/visiosoft/advs-module/src/Adv/AdvRepository.php:402]: we need to use common method for it. it's why we have model and repo

**Review by diashalabi — APPROVED** [review]

**Review by fatihalp — APPROVED** [review]

**Pull request merged**